fastapi/openapi/utils.py

```diff
@@ -296,11 +297,9 @@ def get_openapi_path(
                     deep_dict_update(openapi_response, process_response)
                     openapi_response["description"] = description
             http422 = str(HTTP_422_UNPROCESSABLE_ENTITY)
-            if (all_route_params or route.body_field) and not any(
-                [
-                    status in operation["responses"]
-                    for status in [http422, "4XX", "default"]
-                ]
+            if (((all_route_params or route.body_field))) and all(
+                status not in operation["responses"]
+                for status in [http422, "4XX", "default"]
             ):
```

Comment:
You can also encapsulate the conditions by a util function which will make it easier to understand the conditions by the function name and provide reusability.